### PR TITLE
Generate .rs code for Dali on each main merge # NX-11 #NX-45

### PR DESCRIPTION
**Pull request by obsessed-cake**
## Issues
https://app.clickup.com/t/20465559/NX-11
https://app.clickup.com/t/20465559/NX-45

### CODE REVIEW

**Comment by KaiserKarel** [comment]
Task linked: [NX-11 Generate .rs code for Dali on each main merge](https://app.clickup.com/t/20465559/NX-11)

**Review by KaiserKarel — APPROVED** [review]

**Review by vimukthi-git — APPROVED** [review]

File: code/utils/composable-subxt/dali-subxt-client.nix [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    mkdir -p $out/tmp
+
+    ${pkgs.lib.meta.getExe packages.centauri-codegen} \
+      --path $out/tmp \
+      --parachain-wasm=${packages.dali-runtime}/lib/runtime.optimized.wasm \
+      --relaychain-wasm=${packages.rococo-wasm-runtime}/lib/rococo_runtime.compact.compressed.wasm
+
+    for file in $(ls $out/tmp); do
+      cat $out/tmp/$file | rustfmt --edition=2018 --emit=stdout > $out/$file
+    done
```
Comment (benluelo) [inline code/utils/composable-subxt/dali-subxt-client.nix:21]: any reason why this isn't just `rustfmt $(ls $out/tmp)`?

Comment (benluelo) [reply code/utils/composable-subxt/dali-subxt-client.nix:21]: or `rustfmt $out/tmp/*`

Comment (obsessed-cake) [reply code/utils/composable-subxt/dali-subxt-client.nix]: yes, you are right. we can simplify this.
fixed.

**Comment by obsessed-cake** [comment]
@Mergifyio refresh

File: .github/workflows/build-test-deploy.yml [hunk 2]
```diff
@@ -823,6 +823,26 @@ jobs:
...
           token: "${{ secrets.CACHIX_AUTH_TOKEN }}"
 
+  package-dali-subxt-client:
+    name: "Package Dali subxt client"
+    runs-on:
+      - self-hosted
+      - linux
+      - x64
+      - sre
+    if: github.event_name == 'push' && github.ref == 'refs/heads/main'
```
Comment (KaiserKarel) [inline .github/workflows/build-test-deploy.yml:833]: Shouldn't this be removed? Else we can merge code that breaks this flow?

Comment (obsessed-cake) [reply .github/workflows/build-test-deploy.yml]: The task requires this.
But, you are right.
I removed this.

**Review by KaiserKarel — APPROVED** [review]

File: flake.nix [hunk 3]
```diff
@@ -56,6 +56,7 @@
         ./rust.nix
         ./subwasm.nix
         ./scripts/zombienet/flake-module.nix
+        ./centauri-codegen.nix
```
Comment (dzmitry-lahoda) [inline flake.nix:59]: please move out of root folder, into `.nix` or into `inputs/ComposableFi/centauri/flake-module.nix`

**Review by dzmitry-lahoda — APPROVED** [review]

**Review by dzmitry-lahoda — APPROVED** [review]

**Review by andyjsbell — APPROVED** [review]

**Pull request merged**